server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -1860,20 +1931,42 @@ public void testBlankFieldNamesSubobjectsFalse() throws Exception {
     }
 
     public void testDotsOnlyFieldNames() throws Exception {
-        dotsOnlyFieldNames(createDocumentMapper(mapping(b -> {})));
+        dynamicTrueOrDynamicRuntimeTest(mapper -> {
+            String[] fieldNames = { ".", "..", "..." };
+            for (String fieldName : fieldNames) {
+                MapperParsingException err = expectThrows(
+                    MapperParsingException.class,
+                    () -> mapper.parse(source(b -> b.field(fieldName, "bar")))
+                );
+                assertThat(err.getCause(), notNullValue());
+                assertThat(err.getCause().getMessage(), containsString("field name cannot contain only dots"));
+            }
+        });
     }
 
     public void testDotsOnlyFieldNamesSubobjectsFalse() throws Exception {
-        dotsOnlyFieldNames(createDocumentMapper(topMapping(b -> b.field("subobjects", false))));
+        DocumentMapper mapper = createDocumentMapper(mappingNoSubobjects(b -> {}));
+        String[] fieldNames = { ".", "..", "..." };
+        for (String fieldName : fieldNames) {
+            MapperParsingException err = expectThrows(
+                MapperParsingException.class,
+                () -> mapper.parse(source(b -> b.field(fieldName, "bar")))
+            );
+            assertThat(err.getCause(), notNullValue());
+            // TODO this is actually accepted in the mappings, shall we revert https://github.com/elastic/elasticsearch/pull/90950 ?
```

Comment:
@romseygeek do you have thoughts here? I am wondering if I should revert #90950 , it went in 8.5 too.

Reply:
I think #90950 is the right behaviour.  When you say 'accepted in mappings', do you mean that it's possible to create a field in the mappings called `..`?  But it is then impossible to index into it because the document parser will throw an error?